fili-core/src/test/groovy/com/yahoo/bard/webservice/web/filters/RoleBasedAuthFilterSpec.groovy

```diff
@@ -15,49 +18,71 @@ import javax.ws.rs.core.SecurityContext
 @Timeout(30)
 class RoleBasedAuthFilterSpec extends Specification {
 
-    JerseyTestBinder jtb
+    private static final SystemConfig SYSTEM_CONFIG = SystemConfigProvider.getInstance()
+    private static final String userRolesSetting = SYSTEM_CONFIG.getPackageVariableName("user_roles")
 
-    Class<?>[] getResourceClasses() {
-        [RoleBasedAuthFilterSpec.class, DataServlet.class]
-    }
+    // Memorize pre-test setting value
+    Optional<String> oldUserRoles
+
+    ObjectMapper objectMapper
+    ContainerRequestContext containerRequestContext
+    SecurityContext securityContext
+    RoleBasedAuthFilter roleBasedAuthFilter
 
     def setup() {
-        // Create the test web container to test the resources
-        jtb = new JerseyTestBinder(getResourceClasses())
+        objectMapper = Mock(ObjectMapper)
+        containerRequestContext = Mock(ContainerRequestContext)
+        securityContext = Mock(SecurityContext)
+        containerRequestContext.getSecurityContext() >> securityContext
+        roleBasedAuthFilter = new RoleBasedAuthFilter(objectMapper)
+        
+        try {
+            oldUserRoles = Optional.of(SYSTEM_CONFIG.getStringProperty(userRolesSetting))
+        } catch(SystemConfigException e) {
+            oldUserRoles = Optional.empty()
+        }
+        SYSTEM_CONFIG.setProperty(userRolesSetting, "foo,bar");
     }
 
     def cleanup() {
-        // Release the test web container
-        jtb.tearDown()
+        // Restore pre-test setting value if exists
+        oldUserRoles.ifPresent({userRoles -> SYSTEM_CONFIG.resetProperty(userRolesSetting, userRoles)})
```

Comment:
This needs to clear the setting if the `Optional` is empty, which this doesn't do.